frontend/src/components/ClusterModal.js

```diff
@@ -410,7 +437,9 @@ const ClusterModal = ({
                 Selected {selectedCluster.length - Object.values(excludedImageIndexes).filter(Boolean).length}{" "}
                 of {selectedCluster.length} images (thumbnails: {imageGridSizes.map((size, i) =>
                   <>
-                    <a key={i} href="#" className="text-secondary" onClick={(e) => setImageGridSize(size, e)}>{size.label}</a>
+                    <a key={size.label} href="#" className="text-secondary" onClick={(e) => setImageGridSize(size, e)}>
```

Comment:
I use size.label as key in ClusterModal.

Warning: I can't be 100% sure this is okay without seeing the real data. I assume the label is unique